python/pynessie/commands/contents/commit.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Contents Commit Command CLI."""
+from typing import List, Optional
+
+import click
+
+from ...cli_common_context import ContextObject
+from ...client import NessieClient
+from ...decorators import error_handler, pass_client, validate_reference
+from ...error import NessieNotFoundException
+from ...model import Contents, ContentsSchema, Delete, Operation, Put
+from ...utils import contents_key, format_key
+
+
+@click.command("commit")
+@click.option("-r", "--ref", help="Branch to commit content to. If not supplied the default branch from config is used.")
+@click.option("-i", "--stdin", is_flag=True, help="Read contents for --set from STDIN (separated by Ctrl-D)")
+@click.option(
+    "-s",
+    "--expect-same-contents",
+    is_flag=True,
+    help="Send the same contents both as the new and expected (old contents) parameters for the edit operations",
+)
+@click.option(
+    "-c",
+    "--condition",
+    "expected_hash",
+    required=True,
+    help="Expected hash. Only perform the action if the branch currently points to the hash specified by this option.",
+)
+@click.option("-d", "--delete", is_flag=True, help="Delete a content.")
```

Comment:
`commit -d` looks kind of odd to me... makes me think about the commit itself being deleted. How about `commit --remove-key` (with `-R` for short)?

Reply:
tangential: since the user actually edits the content as JSON, would it make sense to have `nessie checkout` and `nessie commit` as separate operations? The, the user will be able to browse JSON files and do the editing in a more relaxed manner than having to process contents as a stream of edits in random order.

Reply:
Good idea! I will change this

Reply:
Regarding `nessie checkout`, you mean as part of `nessie contents` sub-commands, how will it defer from `nessie contents view`? You mean like writing the actual json files?

Reply:
re: `checkout` I mean actually creating JSON files on disk... certainly not for this PR... future enhancement if people agree :)

Reply:
It doesn't sound like a bad idea to me! I think this can be added under `nessie contents checkout` and in `nessie contents commit`, we can add a flag to use a JSON file instead when committing the content :)

Reply:
> `commit -d` looks kind of odd to me... makes me think about the commit itself being deleted. How about `commit --remove-key` (with `-R` for short)?

I have fixed this :)

Reply:
> tangential: since the user actually edits the content as JSON, would it make sense to have `nessie checkout` and `nessie commit` as separate operations? Then, the user will be able to browse JSON files and do the editing in a more relaxed manner than having to process contents as a stream of edits in random order.

Added https://github.com/projectnessie/nessie/issues/2601